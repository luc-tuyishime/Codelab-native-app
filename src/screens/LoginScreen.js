import React, { Component } from 'react';
import { Text, View, KeyboardAvoidingView, TouchableOpacity } from 'react-native';
import firebase from 'firebase';
import Header from '../components/Header';
import Input from '../components/Input';
import { styles } from '../styles/LoginStyle';


class LoginScreen extends Component {
  state = {
      email: '',
      password: '',
      error: ''
  }

  onButtonPress() {
    this.setState({ error: '', loading: true })
    const { email, password } = this.state;
    firebase.auth().signInWithEmailAndPassword(email, password)
      .then(this.onLoginSuccess.bind(this))
      .catch(() => {
        firebase.auth().createUserWithEmailAndPassword(email, password)
          .then(this.onLoginSuccess.bind(this))
          .catch((error) => {
            let errorCode = error.code
            let errorMessage = error.message;
            if (errorCode == 'auth/weak-password') {
              this.onLoginFailure.bind(this)('Weak password!')
            } else {
              this.onLoginFailure.bind(this)(errorMessage)
            }
          });
      });
  }  
  
  onLoginSuccess() {
    this.setState({
      email: '', password: '', error: '', loading: false
    })
  }  
  
  onLoginFailure(errorMessage) {
    this.setState({ error: errorMessage, loading: false })
  }

  render() {
    return (
    <KeyboardAvoidingView behavior='padding' style={styles.Container}>
     <View style={styles.Container}>
       <View style={styles.infoContainer}>
        <Header title='Codelab'/>
        <Input
        placeholder="Enter email..."
        placeholderTextColor="#fff"
        value={this.state.email}
        keyboardType="email-address"
        returnKeyType="next"
        autoCorrect={false}
        onChangeText={email => this.setState({ email })} 
        />
        <Input style={styles.input} 
        placeholder="Enter password..."
        placeholderTextColor="#fff"
        value={this.state.password}
        returnKeyType="go"
        secureTextEntry={true}
        autoCorrect={true}
        onChangeText={password => this.setState({ password })} 
        />
        <TouchableOpacity style={styles.buttonContainer} onPress={this.onButtonPress.bind(this)}>
         <Text style={styles.buttonText}>Sign In</Text>
        </TouchableOpacity>
       </View>
       <Text style={styles.errorTextStyle}>
        {this.state.error}
        </Text>
      </View>
    </KeyboardAvoidingView> 
    );
  }
}



export default LoginScreen;
