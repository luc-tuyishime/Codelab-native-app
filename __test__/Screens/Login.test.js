import React from 'react';
import { shallow } from 'enzyme';
import LoginScreen from '../../src/screens/LoginScreen';

describe('LOGIN SCREEN', () => {
  describe('render the login page', () => {
    it('should match to snapshot', () => {
      const component = shallow(<LoginScreen />);
      expect(component).toMatchSnapshot();
    });
  });
});