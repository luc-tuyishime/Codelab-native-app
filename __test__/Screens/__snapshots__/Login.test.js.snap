// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LOGIN SCREEN render the login page should match to snapshot 1`] = `
<KeyboardAvoidingView
  behavior="padding"
  enabled={true}
  keyboardVerticalOffset={0}
  style={
    Object {
      "backgroundColor": "rgb(32, 53, 70)",
      "color": "#fff",
      "flex": 1,
      "fontSize": 30,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgb(32, 53, 70)",
        "color": "#fff",
        "flex": 1,
        "fontSize": 30,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 25,
          "padding": 20,
        }
      }
    >
      <Header
        title="Codelab"
      />
      <Input
        autoCorrect={false}
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Enter email..."
        placeholderTextColor="#fff"
        returnKeyType="next"
        value=""
      />
      <Input
        autoCorrect={true}
        onChangeText={[Function]}
        placeholder="Enter password.."
        placeholderTextColor="#fff"
        returnKeyType="go"
        secureTextEntry={true}
        value=""
      />
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "#fff",
            "borderRadius": 10,
            "paddingVertical": 15,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "rgb(32,53,70)",
              "fontSize": 18,
              "fontWeight": "bold",
              "textAlign": "center",
            }
          }
        >
          Sign In
        </Text>
      </TouchableOpacity>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "color": "red",
          "fontSize": 18,
        }
      }
    />
  </View>
</KeyboardAvoidingView>
`;
